docs/pipelineruns.md

```diff
@@ -479,6 +479,24 @@ If used with this `Pipeline`,  `build-task` will use the task specific `PodTempl
 `PipelineTaskRunSpec` may also contain `StepOverrides` and `SidecarOverrides`; see
 [Overriding `Task` `Steps` and `Sidecars`](./taskruns.md#overriding-task-steps-and-sidecars) for more information.
 
+An optional `Metadata` field can be added for each task to have the flexibility to add annotations and labels required in a specific running context.
+
+An example for rendering needed secrets with Vault:
+
+```yaml
+spec:
+  pipelineRef:
+    name: pipeline-name
+  taskRunSpecs:
+    - pipelineTaskName: task-name
+      metadata: 
+        annotations:
+          vault.hashicorp.com/agent-inject-secret-foo: "/path/to/foo"
+          vault.hashicorp.com/role: role-name
+```
+
+The specified metadata will have a higher precedence to keep its value than the metadata added in `PipelineRun.metadata.*` and `Pipeline.spec.tasks.taskSpec.metadata.*` if they hold a same key value.
```

Comment:
nit: this could be phrased as: "If the same key is present in PipelineRun.metadata and Pipeline.spec.tasks.taskSpec.metadata, the value from Pipeline.spec.tasks.taskSpec.metadata will be used".

Reply:
there are 3 metadata mentioned for this case.
think perhaps could follow the direction and rephrase as:
"If the same key is present, the value will be kept by the precedence order as `PipelineRun.spec.taskRunSpec.metadata` > `PipelineRun.metadata` > `Pipeline.spec.tasks.taskSpec.metadata`."